go/arrow/array/builder.go

```diff
@@ -349,12 +349,10 @@ func NewBuilder(mem memory.Allocator, dtype arrow.DataType) Builder {
 		typ := dtype.(*arrow.LargeListViewType)
 		return NewLargeListViewBuilderWithField(mem, typ.ElemField())
 	case arrow.EXTENSION:
-		typ := dtype.(arrow.ExtensionType)
-		bldr := NewExtensionBuilder(mem, typ)
-		if custom, ok := typ.(ExtensionBuilderWrapper); ok {
-			return custom.NewBuilder(bldr)
+		if custom, ok := dtype.(CustomExtensionBuilder); ok {
+			return custom.NewBuilder(mem)
 		}
-		return bldr
+		return NewExtensionBuilder(mem, dtype.(arrow.ExtensionType))
```

Comment:
Can you ensure this is being tested?

Reply:
Yes, good idea. I just added some additional tests comparing the behavior of an extension type with/without a custom builder to really make the distinction clear.